src/hotspot/os/linux/os_linux.cpp

```diff
@@ -401,7 +401,7 @@ bool os::Linux::get_tick_information(CPUPerfTicks* pticks, int which_logical_cpu
 // Returns the kernel thread id of the currently running thread. Kernel
 // thread id is used to access /proc.
 pid_t os::Linux::gettid() {
-  int rslt = syscall(SYS_gettid);
+  int64_t rslt = syscall(SYS_gettid);
```

Comment:
I have to prove I'm not an idiot here :). This is from the Linux manpage:
```
NAME
       syscall - indirect system call

SYNOPSIS
       #define _GNU_SOURCE         /* See feature_test_macros(7) */
       #include <unistd.h>
       #include <sys/syscall.h>   /* For SYS_xxx definitions */

       int syscall(int number, ...);
```
But yes the definition in unistd.h is `long int'.

In platform specific code such as this, when a function returns a long then perhaps we should type the result variable as a long as well, rather than forcing to a 64-bit type?

Reply:
We discussed purging 'long' from the code but since this is platform specific code, then we can use long here to match the type of syscall.